File: tests/testdata/codegen/logical-name-pp/nodejs/logical-name.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as random from "@codeinfra/random";

export = async () => {
    const config = new codeinfra.Config();
    const configLexicalName = config.require("cC-Charlie_charlie.😃⁉️");
    const resourceLexicalName = new random.RandomPet("aA-Alpha_alpha.🤯⁉️", {prefix: configLexicalName});
    return {
        "bB-Beta_beta.💜⁉": resourceLexicalName.id,
        "dD-Delta_delta.🔥⁉": resourceLexicalName.id,
    };
}

File: tests/testdata/codegen/logical-name-pp/python/logical-name.py
import codeinfra
import codeinfra_random as random

config = codeinfra.Config()
config_lexical_name = config.require("cC-Charlie_charlie.😃⁉️")
resource_lexical_name = random.RandomPet("aA-Alpha_alpha.🤯⁉️", prefix=config_lexical_name)
codeinfra.export("bB-Beta_beta.💜⁉", resource_lexical_name.id)
codeinfra.export("dD-Delta_delta.🔥⁉", resource_lexical_name.id)
